pkg/gadgets/internal/test/helpers.go

```diff
@@ -0,0 +1,153 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/google/go-cmp/cmp"
+	"golang.org/x/sys/unix"
+)
+
+// CreateMntNsFilterMap creates and fills an eBPF map that can be used
+// to filter by mount namespace in the different tracers.
+func CreateMntNsFilterMap(t *testing.T, mountNsIDs ...uint64) *ebpf.Map {
+	t.Helper()
+
+	const one = uint32(1)
+
+	mntnsSpec := &ebpf.MapSpec{
+		Name:       "filter_map_" + t.Name(),
+		Type:       ebpf.Hash,
+		KeySize:    8,
+		ValueSize:  4,
+		MaxEntries: 1024,
+	}
+	m, err := ebpf.NewMap(mntnsSpec)
+	if err != nil {
+		t.Fatalf("Failed to create eBPF map: %s", err)
+	}
+
+	for _, mountnsid := range mountNsIDs {
+		if err := m.Put(mountnsid, one); err != nil {
+			m.Close()
+			t.Fatalf("Failed to update eBPF map: %s", err)
+		}
+	}
+
+	return m
+}
+
+// RequireRoot skips the test if the not running as root
+func RequireRoot(t *testing.T) {
+	t.Helper()
+	if unix.Getuid() != 0 {
+		t.Skip("Test requires root")
+	}
+}
+
+type Tracer interface {
+	Stop()
+}
+
+type Test[Event any] struct {
+	GetTracer     func(*Info, func(event Event)) Tracer
+	RunnerConfig  *Config
+	GenerateEvent func() error
+	// One of the returned events should match this
+	GetAnyExpectedEvent func(*Info) *Event
+	// Only one event should be returned and should match this
+	GetExpectedEvent func(*Info) *Event
+}
+
+// Run runs a series of test cases using a dedicated runner.
+func Run[Event any](t *testing.T, tests map[string]Test[Event]) {
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			events := []Event{}
+			eventCallback := func(event Event) {
+				events = append(events, event)
+			}
+
+			runner, err := newRunner(test.RunnerConfig)
+			if err != nil {
+				t.Fatalf("%q: failed to create runner: %s", name, err)
```

Comment:
Now that you use `t.Run`, there is no need to pass around `name` here and below.